server/src/document/DbtTextDocument.ts

```diff
@@ -201,14 +201,20 @@ export class DbtTextDocument {
   }
 
   async resendDiagnostics(): Promise<void> {
-    if (this.destinationState.contextInitialized && this.dbt.dbtReady && !this.currentDbtError && !this.modelCompiler.compilationInProgress) {
+    if (
+      this.destinationState.contextInitialized &&
+      this.dbt.dbtReady &&
+      !this.currentDbtError &&
+      !this.modelCompiler.compilationInProgress &&
+      this.compiledDocument.getText() !== this.rawDocument.getText()
+    ) {
       await this.updateDiagnostics();
     }
   }
 
   debouncedCompile = debounce(async () => {
     this.progressReporter.sendStart(this.rawDocument.uri);
-    await this.modelCompiler.compile(this.getModelPathOrFullyQualifiedName());
+    await this.modelCompiler.compile(this.getModelPathOrFullyQualifiedName(), this.rawDocument.getText().trim().length > 0);
```

Comment:
It is a bit unclear what `this.rawDocument.getText().trim().length > 0` means. Could you please create some method to calculate if fallback is allowed?